adt-pulse-mqtt/server.js

```diff
@@ -32,7 +32,7 @@ client.on("connect", function () {
   console.log("MQTT Sub to: " + alarm_command_topic);
   client.subscribe(alarm_command_topic);
   if (smartthings) {
-    client.subscribe(smartthings_topic + "/ADT Alarm System/alarm/state");
+    client.subscribe(smartthings_topic + "/security/ADT Alarm System/state");
```

Comment:
The MQTT Discovery driver does not currently support `/security/` devices but I'm leaving this here for future support.  At worst, it is currently ignored by the MQTT Discovery edge driver.  Only the following devices are handled by the driver:

- `/alarm` -> siren
- `/motion` -> motion sensor
- `/contact` -> contact (open/close) sensor

I will continue working with the community to see if we can get `/security` supported on the driver for security panel support (home, away, etc.).  We can also work to translate `/alarm` as siren status if you know how to scrape that information from the ADT Pulse site.  I don't have the stomach to trigger my siren and simultaneously try to deconstruct how that is exposed via Pulse. 😄

Reply:
Lemme know when this is available as an updated Docker image and I'll test it out.

Reply:
It actually supports spaces just fine. I have no idea why the documentation incorrectly states that.

Reply:
This is also something I noted in my open issue with them and to the community. Neither MQTT nor the driver itself has the slightest issue supporting spaces.

Reply:
Note that I have an active setup including SmartThings using this edge driver and the code with no issues. I've been burning it in for a while now.

Reply:
Lemme take a look.  Maybe I messed something up when I was rebasing against your prettier updates.

Reply:
I think I figured it out.  For whatever reason, the edge driver does _not_ like the `/security/` type being in there specifically related to the `/config` portion.  It ends up causing the whole thing to not work for unknown reasons.  I never noticed this because, once the devices are created, it doesn't seem to care about the `/state` updates being there.  We'll have to remove the Security Panel support from the `/smartthings` topic for now in order to get this working.

Reply:
BTW, just for the heck of it I tested with spaces replaced as `_` and that didn't help in the least.

Reply:
What I did for now is push those security panel MQTT topics to `/smartthings_future/security` instead.  That got them off the `/smartthings/` topic and the MQTT Edge driver immediately re-detected my devices and set them up again.  Even with spaces.

Reply:
We may also need to do some more work on the `.../config` part of this.  The reason is that the MQTT edge driver uses the `/config` topic to be told that there's a new device.  The problem is it also specifically says _not_ to set those topics as "retain" on MQTT.  This sets us up for a bit of a race condition:

1. The `adt-pulse-mqtt` bridge is started up first. It detects new devices and fires off the `/config` as a "one-shot" with no retention.
2. The MQTT Discovery driver is brought online and subscribes to the topics.

At this point, it will never get the `/config` topics unless you restart `adt-pulse-mqtt`.  What we'd need to do to avoid this is one of:

- Retain those `/config` messages on MQTT
- Periodically fire off new `/config` messages even though we've already done it previously.
- Update the documentation so users understand they need to bring up and subscribe the MQTT Discovery driver _before_ bringing up `adt-pulse-mqtt`.  Or restart it after they do.

Reply:
I will put up an MR in a couple minutes.

Reply:
https://github.com/BigThunderSR/adt-pulse-mqtt/pull/210